src/helpers/manage-issue-due-dates.ts

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2023 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { OVERDUE_ISSUE, ALMOST_OVERDUE_ISSUE, PRIORITY_1, PRIORITY_2, PRIORITY_3, PRIORITY_4 } from '../constants';
+import { HelperInputs } from '../types/generated';
+import { context } from '@actions/github';
+import { octokit } from '../octokit';
+import { paginateAllOpenIssues } from '../utils/paginate-open-issues';
+import { paginateAllCommentsOnIssue } from '../utils/paginate-comments-on-issue';
+import { CommentList, SingleComment, IssueLabels, IssueList } from '../types/github';
+
+type PriorityLabelType = {
+  priority_1: string;
+  priority_2: string;
+  priority_3: string;
+  priority_4: string;
+};
+
+export class AddOverdueIssueLabel extends HelperInputs {
+  /* The threshold (in days) of when to apply the "due soon" label
+   * @default 7
+   */
+  warningThreshold?: number;
+  almostOverdueLabel?: string;
+  overdueLabel?: string;
+  priorityLabels?: PriorityLabelType;
```

Comment:
These all need to be strings since actions inputs are all strings. Also run the `npm run generate` script to add generated types for these inputs, as well as adding them to `action.yml`